### PR TITLE
Improve data stream parsing timestamp field error message

### PR DESCRIPTION
**Pull request by weizijun**
When @timestamp is not found,  the error message is :
`Failed to parse object: expecting token of type [START_OBJECT] but found [null]`

Users may don't know the exact failed reason. The PR is to improve this case. And add a @timestamp not found case.

### CODE REVIEW

**Comment by martijnvg** [comment]
@elasticmachine test this please

File: docs/changelog/84391.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+pr: 84391
```
Comment (martijnvg) [inline docs/changelog/84391.yaml:1]: @weizijun I don't think these changelog entries should be added for time series project? This feature is currently behind a feature flag and not ready yet to be used in production. The entire time series feature will likely have its own changelog entry when it is ready.

Comment (weizijun) [reply docs/changelog/84391.yaml:1]: done

**Comment by weizijun** [comment]
@elasticmachine update branch

**Comment by martijnvg** [comment]
@elasticmachine test this please

**Review by martijnvg — APPROVED** [review]
Thanks @weizijun.

**Pull request merged**